plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java

```diff
@@ -131,6 +131,26 @@ public static Library collectLibraryFiles(IJavaProject javaProject, List<Library
       subMonitor.worked(1);
     }
 
+    // need to get old master library entries first if they exist
+    try {
+      LibraryClasspathContainerSerializer serializer = new LibraryClasspathContainerSerializer();
+      List<String> previouslyAddedLibraries = serializer.loadLibraryIds(javaProject, null);
+      for (String id : previouslyAddedLibraries) {
+        Library library = CloudLibraries.getLibrary(id);
+        if (library != null) { // happens mostly in tests but could also be null
```

Comment:
Would be nice to check `!dependentIds.contains(library.getId())` too, because otherwise, the master container JSON file can have duplicate library IDs.
```
[
  "googlecloudstorage",
  "googlecloudspanner",
  "googlecloudstorage",
  "googlecloudstorage",
  "googlecloudstorage",
  "googlecloudstorage",
  "googlecloudpubsub",
  "googlecloudpubsub"
]
```
This is not a critical issue though, as duplicate IDs don't affect correct functionality.

Reply:
done